core/src/main/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationService.scala

```diff
@@ -127,4 +128,30 @@ class GoogleProjectRegistrationService(protected val ctx: RawlsRequestContext,
       // 3. Delete the record in the GOOGLE_PROJECT table.
       _ <- googleProjectRegRepo.deleteGoogleProjectRegistration(googleProjectId)
     } yield ()
+
+  def getGoogleProjects(billingProjectName: Option[RawlsBillingProjectName],
+                        pageSize: Int,
+                        offset: Int
+  ): Future[Seq[GoogleProjectRegistration]] = {
+    val accessibleGoogleProjectsFuture = samDAO
+      .listResourcesWithActions(SamResourceTypeNames.googleProject, SamGoogleProjectActions.read, ctx)
+      .map(_.map(resource => GoogleProjectId(resource.getResourceId)).toSet)
+
+    accessibleGoogleProjectsFuture.flatMap { accessibleGoogleProjects =>
+      googleProjectRegRepo.getGoogleProjectRegistrations(accessibleGoogleProjects, billingProjectName, pageSize, offset)
+    }
```

Comment:
maybe a little nicer as a for comprehension, your choice
```suggestion
    for {
      accessibleGoogleProjectResources <- samDAO
        .listResourcesWithActions(SamResourceTypeNames.googleProject, SamGoogleProjectActions.read, ctx)

      accessibleGoogleProjects = accessibleGoogleProjectResources.map(resource => GoogleProjectId(resource.getResourceId)).toSet
      
      projectRegistrations <- googleProjectRegRepo.getGoogleProjectRegistrations(accessibleGoogleProjects, billingProjectName, pageSize, offset)
    } yield projectRegistrations
```